### DISCUSSION TITLE
After upgrade to Mn 3.0.0-M1/2 -> NonUniqueBeanException

### DISCUSSION THREAD

**Discussion by martisaw** [micronaut-projects/micronaut-core#5636] [General] [answered]
Hi,

we ([micronaut-camunda-bpm project](https://github.com/NovatecConsulting/micronaut-camunda-bpm)) want to upgrade to the newest Milestone-Release of Micronaut. However, we discovered that some tests are now throwing NonUniqueBeanExceptions. 

We want to retrieve Beans (where we don't know the type yet) from the applicationContext with `applicationContext.getBean(Object.class, Qualifiers.byName(".."));`
Using Micronaut 2.5.x, this worked fine. With 3.0.0-M1 and M2, we are getting the following error message: `Multiple possible bean candidates found: [com.example.SampleSingleton, io.micronaut.context.BeanProvider, jakarta.inject.Provider]`

From our point of view this is an unwanted change of behaviour. From an application developer point of view it is strange that the BeanProviders and Providers inherit the qualifier. What do you think?

See here for a sample project: https://github.com/martisaw/mn-nonUniqueBean

**Comment by graemerocher** [comment, marked as answer]
Can you turn this into an issue report? Seems like a bug worthy of fixing
